### PR TITLE
[asdf] Do not copy array data

### LINKED ISSUE #454: xarray.Variable deserialization casts NDArrayType (asdf) to np.ndarray
As a consequence, memory mapping of stored arrays does not work, e.g. copy_arrays flag has no effect.

### PR DESCRIPTION
**Pull request by marscher**
## Changes

Because asdf does not handle time datatypes (correctly), there was a line of code within the serialization of xarray Variable to convert all ndarray like types from the asdf ndarray wrapper. This effectively prevents the use of the "copy_arrays" flag, since the
data got always copied.

## Related Issues

Closes #454

## Checks

- [ ] updated CHANGELOG.md
- [ ] updated tests
- [ ] updated doc/
- [ ] update example/tutorial notebooks

### CODE REVIEW

File: weldx/asdf/tags/weldx/core/common_types.py [hunk 1]
```diff
@@ -134,7 +134,12 @@ def from_tree(cls, tree, ctx):
 
         """
         dtype = np.dtype(tree["dtype"])
-        data = np.array(tree["data"], dtype=dtype)
+        # TODO: it would be ideal, if asdf would handle time types natively.
```
Comment (CagtayFabry) [inline weldx/asdf/tags/weldx/core/common_types.py:137]: cref https://github.com/asdf-format/asdf-standard/issues/270

**Comment by CagtayFabry** [comment]
nice, does this work now with `copy_arrays`?

we should probably dig into the (asdf related) code for more `np.array()`, `np.asarray()` etc. calls

File: weldx/asdf/tags/weldx/core/common_types.py (hunk 1 above)
Comment (marscher) [reply weldx/asdf/tags/weldx/core/common_types.py:137]: Thanks for the reference (and welcome back). Is there any roadmap for asdf standard 2.0? I guess this has to be designed super carefully in order not breaking every bodies files.

**Comment by marscher** [comment]
Our unit tests switch the copy_arrays flag at least. But I doubt that the underlying functionality of mem mapping is really being tested. We can argue that this is already tested extensively upstream.

**Comment by CagtayFabry** [comment]
> Our unit tests switch the copy_arrays flag at least. But I doubt that the underlying functionality of mem mapping is really being tested. We can argue that this is already tested extensively upstream.

that is certainly not tested, we just use these parameters to catch any errors that might flare up
we would probably have to look into a memory profiler to see the difference

File: weldx/asdf/tags/weldx/core/common_types.py (hunk 1 above)
Comment (CagtayFabry) [reply weldx/asdf/tags/weldx/core/common_types.py:137]: haven't looked into the state of the asdf 2.0 standard for a while tbh

**Review by CagtayFabry — APPROVED** [review]

**Comment by marscher** [comment]
I got my memory profiler booted up anyways for another PR, so eventually I'll take a look.

**Pull request merged**